src/context.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::HashMap;
+
+use serde::Serialize;
+use windows_core::{BSTR, VARIANT};
+
+use crate::{WMIConnection, WMIResult};
+
+#[derive(Debug, PartialEq, Serialize, Clone)]
+#[serde(untagged)]
+pub enum ContextValueType {
```

Comment:
Maybe add to this a `non_exhaustive` attribute, so adding more options won't be a breaking change

Reply:
We can also accept replace `ContextValueType` with `Variant` / `impl Into<Variant>`, but then it'll be better to have `impl_try_from_variant` generate both impls (the new one being `impl From<$target_type> for Variant`)

Reply:
👍 about the doc - this makes a lot of sense.